modules/product/src/Plugin/Field/FieldWidget/ProductVariationAttributesWidget.php

```diff
@@ -357,7 +357,12 @@ protected function getAttributeValues(array $variations, $field_name, callable $
    * Ajax callback.
    */
   public static function ajaxRefresh(array $form, FormStateInterface $form_state) {
-    return $form;
+    /** @var \Drupal\Core\Ajax\AjaxResponse $response */
+    $response = \Drupal::service('main_content_renderer.ajax')->renderResponse($form, \Drupal::service('request_stack')->getCurrentRequest(), \Drupal::service('current_route_match'));
+
+    // Allow other modules to add arbitrary AJAX commands on the refresh.
+    \Drupal::moduleHandler()->alter('commerce_product_attributes_refresh', $response, $form, $form_state);
```

Comment:
I guessed so, but how should an injected service be called on a static function? No instance, no injected service ;)